File: models/page.go
package models

import (
	"fmt"
	"html/template"
	"strings"
)

//Page type contains page info
type Page struct {
	ID              uint   `json:"id"`
	Name            string `json:"name" binding:"required"`
	Slug            string `json:"slug"`
	Content         string `json:"content"`
	Published       bool   `json:"published"`
	MetaDescription string `json:"meta_description"`
	MetaKeywords    string `json:"meta_keywords"`
}

func (page *Page) URL() string {
	return fmt.Sprintf("/page/%d-%s", page.ID, page.Slug)
}

func (page *Page) EditURL() string {
	return fmt.Sprintf("/admin#/pages/%d", page.ID)
}

func (page *Page) HTMLContent() template.HTML {
	return template.HTML(page.Content)
}

func (page *Page) BeforeCreate() (err error) {
	if strings.TrimSpace(page.Slug) == "" {
		page.Slug = createSlug(page.Name)
	}
	return
}

func (page *Page) BeforeSave() (err error) {
	if strings.TrimSpace(page.Slug) == "" {
		page.Slug = createSlug(page.Name)
	}
	return
}

File: assets/js/page/page.js
'use strict';


module.exports = function(data){
  data = data || {};
  this.id = m.prop(data.id || 0);
  this.name = m.prop(data.name || '');
  this.slug = m.prop(data.slug || '');
  this.content = m.prop(data.content || '');
  this.published = m.prop(data.published || true);
  this.meta_description = m.prop(data.meta_description || '');
  this.meta_keywords = m.prop(data.meta_keywords || '');
}
